### DISCUSSION TITLE
Can we fetch state in get query

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#144] [Q&A]
I want to do some validation in get query.. Considering all the business validation should happen on state.. 

Could you suggest how can I fetch the state details for validation in get query ?

**Comment by alexeyzimarev** [comment]
The sample project has an API endpoint that exposes the aggregate state. I am not sure what the issue is, could you elaborate?

**Reply by alexeyzimarev** [reply]
I can also add that the _business_ validation (I don't like the term) happens inside the domain model, where you have access to the state.
